src/libPMacc/include/algorithms/math/floatMath/floatingPoint.tpp

```diff
@@ -60,6 +87,32 @@ struct Float2int_rd<float>
     }
 };
 
+template<>
+struct Float2int_rn<float>
+{
+    typedef int result;
+
+    HDINLINE result operator( )(float value)
+    {
+#if __CUDA_ARCH__
+        return ::__float2int_rn( value );
+#else
+        if(value < 0.0f)
+            return -(*this)(-value);
+        /* Round towards nearest with x.5 rounded to +inf but take care of
+         * floating point precision (e.g. 8388609.0 + 0.5 == 8388610.0) */
+        float intPart;
+        float fracPart = std::modf(value, &intPart);
+        result res = float2int_rd(fracPart + 0.5f) + static_cast<int>(intPart);
+        /* If we were close to x.5 then make sure res is even */
+        if( abs(0.5f - (res - value)) < std::numeric_limits<float>::epsilon() )
```

Comment:
Offline discussion with @psychocoderHPC: It is better to use `fracPart` instead of `(res - value)`